plugins/recall/scripts/session-index.py

```
@@ -0,0 +1,565 @@
+    # A build holds one write transaction for the whole pass; the default 5s
+    # busy timeout expires when two SessionEnd hooks fire at once, killing the
+    # loser with "database is locked".
+    if fresh:
+        # The index aggregates every prompt typed on this host into one file —
+        # a higher-value target than the transcripts it is built from. Created
+        # 0600 BEFORE sqlite opens it: a chmod afterwards leaves a readable
+        # window, which is the exact reasoning the failure log already applied.
+        try:
+            os.close(os.open(DB_PATH, os.O_CREAT | os.O_EXCL | os.O_WRONLY, 0o600))
+        except OSError:
+            pass
```

Ruff (0.16.1)

[warning] Use `contextlib.suppress(OSError)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(OSError): ...`

(SIM105)